### PR TITLE
Feature/unit tests/infra 30039/event ingestors

### PR DESCRIPTION
**Pull request by kkania-splunk**
unit tests for event ingestors

### CODE REVIEW

File: tests/unit/test_event_ingestors/conftest.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
+import pytest
+from recordtype import recordtype
+from .test_hec_event_metric_raw_ingestor import HEC_URI
+
+SampleEvent = recordtype(
```
Comment (uoboda-splunk) [inline tests/unit/test_event_ingestors/conftest.py:5]: 👍 +1

File: tests/unit/test_event_ingestors/test_hec_event_metric_raw_ingestor.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        ("hec_raw_event_ingestor", f"{HEC_URI}/raw", True),
+    ],
+)
+def test_exception_raised_when_event_ingestion_returns_error(
+    requests_mock, ingestors, caplog, ingestor_name, url, check_logger
+):
+    requests_mock.post(url, text="Not Found", status_code=404)
+    ingestor = ingestors[ingestor_name]
+    pytest.raises(Exception, ingestor["ingestor"].ingest, *(ingestor["events"], 1))
+    if check_logger:
```
Comment (uoboda-splunk) [inline tests/unit/test_event_ingestors/test_hec_event_metric_raw_ingestor.py:101]: Maybe using `match` parameter for pytest.raises? https://docs.pytest.org/en/stable/assert.html#assertraises

File: tests/unit/test_event_ingestors/test_hec_event_metric_raw_ingestor.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        ("hec_metric_event_ingestor", HEC_URI, False),
+        ("hec_raw_event_ingestor", f"{HEC_URI}/raw", True),
+    ],
+)
+def test_exception_raised_when_event_ingestion_returns_error(
+    requests_mock, ingestors, caplog, ingestor_name, url, check_logger
+):
+    requests_mock.post(url, text="Not Found", status_code=404)
+    ingestor = ingestors[ingestor_name]
+    pytest.raises(Exception, ingestor["ingestor"].ingest, *(ingestor["events"], 1))
```
Comment (uoboda-splunk) [inline tests/unit/test_event_ingestors/test_hec_event_metric_raw_ingestor.py:100]: How does it work? I always see `pytest.raises` used as context manager: `with pytest.raises(...`

Comment (kkania-splunk) [reply tests/unit/test_event_ingestors/test_hec_event_metric_raw_ingestor.py]: https://docs.pytest.org/en/stable/assert.html#assertraises
There’s an alternate form of the pytest.raises() function where you pass a function that will be executed with the given `*args` and `**kwargs` and assert that the given exception is raised:
`pytest.raises(ExpectedException, func, *args, **kwargs)`
however it doesn't work with `match`

File: tests/unit/test_event_ingestors/test_hec_event_metric_raw_ingestor.py (hunk 2 above)
Comment (kkania-splunk) [reply tests/unit/test_event_ingestors/test_hec_event_metric_raw_ingestor.py:102]: done

**Review by uoboda-splunk — APPROVED** [review]

**Pull request merged**